warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownLargeFieldedListsVisitor.java

```diff
@@ -81,96 +95,168 @@ public Object visit(ASTOrNode node, Object data) {
+            return rangeNodeToRange(node.jjtGetChild(0));
+        } else if ((node.jjtGetNumChildren() == 2) && node instanceof ASTAndNode) {
+            JexlNode leftChild = node.jjtGetChild(0);
+            JexlNode rightChild = node.jjtGetChild(1);
+            return new Range(new Key(String.valueOf(JexlASTHelper.getLiteralValue(leftChild))), leftChild instanceof ASTGENode, new Key(
+                            String.valueOf(JexlASTHelper.getLiteralValue(rightChild))), rightChild instanceof ASTLENode);
+        } else {
+            return null;
+        }
+    }
+    
     protected boolean isIndexed(String field) {
         return config.getIndexedFields().contains(JexlASTHelper.deconstructIdentifier(field));
     }
     
-    /**
-     * Get the nodes mapped by fieldname. Only equality and negated equalities are included. Anything else is placed into the OTHER bucket.
-     * 
-     * @param nodes
-     * @return the nodes mapped by filename
-     */
-    protected Multimap<String,JexlNode> getNodesByField(JexlNode[] nodes) {
-        Multimap<String,JexlNode> nodeMap = ArrayListMultimap.create();
-        for (JexlNode node : nodes) {
-            nodeMap.put(getEqualityIdentifier(node), node);
-        }
-        return nodeMap;
+    protected void assignNodeByField(JexlNode origNode, Multimap<String,JexlNode> eqNodes, Multimap<String,JexlNode> rangeNodes, List<JexlNode> otherNodes) {
+        assignNodeByField(origNode, origNode, eqNodes, rangeNodes, otherNodes);
     }
     
-    /**
-     * Get the identifier from the equality. If this node is anything else then OTHER_NODES is returned.
-     * 
-     * @param node
-     * @return the node's identifier iff EQ node, otherwise OTHER_NODES.
-     */
-    protected String getEqualityIdentifier(JexlNode node) {
-        if (node instanceof ASTEQNode) {
-            return JexlASTHelper.getIdentifier(node);
-        } else if ((node.jjtGetNumChildren() == 1) && (node instanceof ASTReferenceExpression || node instanceof ASTReference)) {
-            return getEqualityIdentifier(node.jjtGetChild(0));
+    protected void assignNodeByField(JexlNode origNode, JexlNode subNode, Multimap<String,JexlNode> eqNodes, Multimap<String,JexlNode> rangeNodes,
+                    List<JexlNode> otherNodes) {
+        if (subNode instanceof ASTEQNode) {
+            eqNodes.put(JexlASTHelper.getIdentifier(subNode), origNode);
+        } else if (ExceededValueThresholdMarkerJexlNode.instanceOf(subNode)) {
+            assignNodeByField(origNode, ExceededValueThresholdMarkerJexlNode.getExceededValueThresholdSource(subNode), eqNodes, rangeNodes, otherNodes);
+        }
+        // else if (ASTDelayedPredicate.instanceOf(subNode)) {
+        // assignNodeByField(origNode, ASTDelayedPredicate.getDelayedPredicateSource(subNode), eqNodes, rangeNodes, otherNodes);
+        // }
```

Comment:
remove this

Reply:
Whoops!  Good catch.  Removed in the latest commit.